dart/dynamics/Entity.h

```diff
@@ -63,11 +63,23 @@ class Shape;
 /// may have different policies about how/if their reference frame or name
 /// can be changed. Use the Detachable class to create an Entity whose reference
 /// Frame can be changed arbitrarily.
-class Entity
+class Entity : public common::Subscription
 {
 public:
   friend class Frame;
 
+  enum Notification {
```

Comment:
The Signal class looks really good to me. I do plan on doing some multithreading down the road, but I would be using QThreads for that, so I would probably connect Qt signals/slots to the DART signals to facilitate that.

The only question left would be what to name the Subscriber/Publisher classes because I think those names sound too general for something with such a specific purpose.